avsomr/avsimport.cpp

```diff
@@ -0,0 +1,163 @@
+
+#include <QFileInfo>
+
+#include "libmscore/score.h"
+#include "libmscore/text.h"
+
+#include "avslog.h"
+#include "msmrfile.h"
+#include "avsomr.h"
+#include "avsomrreader.h"
+#include "avsomrnetrecognizer.h"
+#include "avsomrlocalrecognizer.h"
+#include "avsomrlocal.h"
+#include "ui/recognitionproccessdialog.h"
+#include "ui/infopopup.h"
+
+namespace Ms {
+
+extern Score::FileError importMusicXml(MasterScore* score, QIODevice* dev, const QString& name);
+
+//---------------------------------------------------------
+//   doImportMSMR
+//     return Score::FileError::FILE_* errors
+//---------------------------------------------------------
+
+static Score::FileError doImportMSMR(MasterScore* score
+                                     , QIODevice *data
+                                     , const QString& filePath
+                                     , bool created) {
+
+      IF_ASSERT(score) {
+            return Score::FileError::FILE_ERROR;
+            }
+
+      IF_ASSERT(data) {
+            return Score::FileError::FILE_ERROR;
+            }
+
+      QString baseName = QFileInfo(filePath).baseName();
+      auto msmr = std::make_shared<Avs::MsmrFile>(data->readAll(), baseName);
+
+      // score
+      {
+      QByteArray msczData = msmr->readMscz();
+      if (!msczData.isEmpty()) {
+
+            // read mscz
+
+            QBuffer msczBuf(&msczData);
+            msczBuf.open(QIODevice::ReadOnly);
+            Score::FileError err = score->loadMsc("score.mscz", &msczBuf, true);
+            if (err != Score::FileError::FILE_NO_ERROR)
+                  return err;
+
+            }
+      else {
+
+            // import MusicXml
+
+            QByteArray muzicXmlData = msmr->readMuzicXml();
+            if (muzicXmlData.isEmpty())
+                  return Score::FileError::FILE_BAD_FORMAT;
+
+            QBuffer muzicXmlBuf(&muzicXmlData);
+            muzicXmlBuf.open(QIODevice::ReadOnly);
+
+            Score::FileError err = importMusicXml(score, &muzicXmlBuf, baseName);
+            if (err != Score::FileError::FILE_NO_ERROR)
+                  return err;
+
+            Text* title = score->getText(Tid::TITLE);
+            if (title)
+                  title->setPlainText(baseName);
+
+            }
+      }
+
+      // omr
+      {
+      QByteArray ormData = msmr->readOmr();
+      if (ormData.isEmpty()) {
+            return Score::FileError::FILE_BAD_FORMAT;
+            }
+
+      QBuffer ormBuf(&ormData);
+      Avs::AvsOmrReader reader;
+      std::shared_ptr<Avs::AvsOmr> omr = reader.read(&ormBuf);
+      if (!omr)
+            return Score::FileError::FILE_BAD_FORMAT;
+
+      omr->setMsmrFile(msmr);
+      score->setAvsOmr(omr);
+      }
```

Comment:
The entire section above up to line 44 needs to get indented one level more. Or that pair of braces removed. Probably better the latter